src/main/java/emu/grasscutter/server/packet/recv/HandlerPlayerForceExitReq.java

```diff
@@ -10,5 +11,7 @@ public class HandlerPlayerForceExitReq extends PacketHandler {
 	@Override
 	public void handle(GameSession session, byte[] header, byte[] payload) throws Exception {
 		// Client should auto disconnect right now
+		session.send(new BasePacket(PacketOpcodes.PlayerForceExitRsp));
+		session.getPlayer().onLogout();
```

Comment:
You dont need to call onLogout, just call session.close() and logout should theoretically call onLogout from it.